### ISSUE TITLE
It would be nice to be able to limit the number of jobs used by `TestTree`s

### ISSUE THREAD

**Issue by michaelpj** [UnkindPartition/tasty#406]
From looking at the code and experimentation, I _believe_ that the `NumThreads` option only takes effect at the top-level. So you can't, for example, limit a particular test tree to be run with only one job (in case the tests interfere with each other).

**Comment by michaelpj** [comment]
And that led me to `ExecutionMode`, which seems like it's exactly what I want but sadly isn't exported. Maybe there's a reason why it's only available through `sequentialTestGroup`, but I'm not sure :thinking:

**Comment by michaelpj** [comment]
Indeed, `sequentialTestGroup` does not do what I want, since it only makes the tests directly in the group run sequentially, whereas tests in nested groups continue to execute in parallel. I really do just want to set the execution mode to sequential for a whole test tree!

**Comment by Bodigrim** [comment]
@michaelpj it is admittedly awkward, but you can pattern-match on 
```haskell
PlusTestOptions mkSequential = sequentialTestGroup "foo" AllFinish []
```
and then
```haskell
defaultMain $ PlusTestOptions mkSequential testTree
```

I'm curious if it works, please tell if you give it a try.

**Comment by dyniec** [comment]
contructors of TestTree aren't exported, so PlusTestOptions is not visible :/

**Comment by Bodigrim** [comment]
@dyniec constructors of `TestTree` are exported from https://hackage.haskell.org/package/tasty-1.5/docs/Test-Tasty-Runners.html#t:TestTree
